Not possible to launch anymore Stirling-PDF

I have tried Stirling-PDF 0.25.0 with the  Stirling-PDF.exe  in a VM under Windows 11, i have stopped the VM and started it later then now i can't launch  Stirling-PDF.

I have opened the log file and it say that : 

`2024-05-29 13:45:02,679 ERROR o.s.b.SpringApplication [main] Application run failed
java.lang.RuntimeException: Failed to initialize application configuration
	at stirling.software.SPDF.config.ConfigInitializer.initialize(ConfigInitializer.java:24)
	at org.springframework.boot.SpringApplication.applyInitializers(SpringApplication.java:627)
	at org.springframework.boot.SpringApplication.prepareContext(SpringApplication.java:400)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:333)
	at stirling.software.SPDF.SPdfApplication.main(SPdfApplication.java:100)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:91)
	at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:53)
	at org.springframework.boot.loader.launch.JarLauncher.main(JarLauncher.java:58)
Caused by: java.nio.file.FileSystemNotFoundException: null
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.getFileSystem(ZipFileSystemProvider.java:156)
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.getPath(ZipFileSystemProvider.java:142)
	at java.base/java.nio.file.Path.of(Path.java:209)
	at java.base/java.nio.file.Paths.get(Paths.java:98)
	at stirling.software.SPDF.config.ConfigInitializer.ensureConfigExists(ConfigInitializer.java:49)
	at stirling.software.SPDF.config.ConfigInitializer.initialize(ConfigInitializer.java:22)
	... 9 common frames omitted
2024-05-29 13:45:58,698 ERROR o.s.b.SpringApplication [main] Application run failed
java.lang.RuntimeException: Failed to initialize application configuration
	at stirling.software.SPDF.config.ConfigInitializer.initialize(ConfigInitializer.java:24)
	at org.springframework.boot.SpringApplication.applyInitializers(SpringApplication.java:627)
	at org.springframework.boot.SpringApplication.prepareContext(SpringApplication.java:400)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:333)
	at stirling.software.SPDF.SPdfApplication.main(SPdfApplication.java:100)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:91)
	at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:53)
	at org.springframework.boot.loader.launch.JarLauncher.main(JarLauncher.java:58)
Caused by: java.nio.file.FileSystemNotFoundException: null
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.getFileSystem(ZipFileSystemProvider.java:156)
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.getPath(ZipFileSystemProvider.java:142)
	at java.base/java.nio.file.Path.of(Path.java:209)
	at java.base/java.nio.file.Paths.get(Paths.java:98)
	at stirling.software.SPDF.config.ConfigInitializer.ensureConfigExists(ConfigInitializer.java:49)
	at stirling.software.SPDF.config.ConfigInitializer.initialize(ConfigInitializer.java:22)
	... 9 common frames omitted
`

Hi all.
I'm new to Stirling-PDF.
Anyway today I installed its Windows version (Stirling-pdf.exe) v.0.25 with JDK 1.7 in a W11 pro (last updates installed) environment.
This is what happened:
1. I create a directory:
C:\Program Files\Stirling-PDF
inside which I put Stirling-PDF.exe.
2. I created a shortcut for the .exe file on the desktop and I set it to run as administrator.
3. The first time I run it from the shortcut it asks me for the permission to run as admin and I give it.
4. A terminal opens and, after a while, also my default browser (Firefox, last version) opens at localhost:8080 showing Stirling-PDF initial page.
Please note that I'm using the version without the login.
5. Then, I close the browser and the terminal session.
6. At this point, if I start the program again, the terminal opens for just a second, then closes abruptly and no browser session starts.
7. If I delete the C:\Program Files\Stirling-PDF\configs\settings.yml file, and I retry to start the program then it works again.
And this happens any time!

Herein below I report the log file (info.log):
---------
2024-05-31 12:11:33,322 ERROR o.s.b.SpringApplication [main] Application run failed
java.lang.RuntimeException: Failed to initialize application configuration
	at stirling.software.SPDF.config.ConfigInitializer.initialize(ConfigInitializer.java:24)
	at org.springframework.boot.SpringApplication.applyInitializers(SpringApplication.java:627)
	at org.springframework.boot.SpringApplication.prepareContext(SpringApplication.java:400)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:333)
	at stirling.software.SPDF.SPdfApplication.main(SPdfApplication.java:100)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:91)
	at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:53)
	at org.springframework.boot.loader.launch.JarLauncher.main(JarLauncher.java:58)
Caused by: java.nio.file.FileSystemNotFoundException: null
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.getFileSystem(ZipFileSystemProvider.java:156)
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.getPath(ZipFileSystemProvider.java:142)
	at java.base/java.nio.file.Path.of(Path.java:209)
	at java.base/java.nio.file.Paths.get(Paths.java:98)
	at stirling.software.SPDF.config.ConfigInitializer.ensureConfigExists(ConfigInitializer.java:49)
	at stirling.software.SPDF.config.ConfigInitializer.initialize(ConfigInitializer.java:22)
	... 9 common frames omitted
------------

Regards,
Topoldo/Leopoldo

I did what you suggested.

First of all, after substituting in the desktop shortcut the command:
`"C:\Program Files\Stirling-PDF\Stirling-PDF.exe"`
with:
`%comspec% /c "C:\Program Files\Stirling-PDF\Stirling-PDF.exe"`
**BOTH** the shortcut and the "original" .exe file changed their Stirling-PDF icon with the Windows "cmd" icon.
Beside, I deleted the
`C:\Program Files\Stirling-PDF\configs\settings.yml`
file and I tried to start Stirling-PDF as it was from scratch.

In this case it worked (as expected when there is no `settings.yml`), ie:
- it opened a terminal window
- it asked for the permission to run as administrator
- after my OK, it closed the first (I suppose `%comspec% /c`) terminal window and opened the "real" (Stirling-PDF) one 
- after a while Firefox opened on page `localhost:8080` and the home page of Stirling-PDF appeared inside it.

The info.log was the following:
```
2024-06-01 09:13:27,965 WARN s.s.S.SPdfApplication [main] External configuration file 'configs/settings.yml' does not exist. Using default configuration and environment configuration instead.
2024-06-01 09:13:28,359 INFO s.s.S.SPdfApplication [main] Starting SPdfApplication v0.25.0 using Java 21.0.3 with PID 10608 (C:\Program Files\Stirling-PDF\Stirling-PDF.exe started by Leopoldo in C:\Program Files\Stirling-PDF)
2024-06-01 09:13:28,360 INFO s.s.S.SPdfApplication [main] No active profile set, falling back to 1 default profile: "default"
2024-06-01 09:13:29,875 INFO o.s.b.w.e.j.JettyServletWebServerFactory [main] Server initialized with port: 8080
2024-06-01 09:13:29,880 INFO o.e.j.s.Server [main] jetty-12.0.7; built: 2024-02-29T21:19:41.771Z; git: c89aca8fd34083befd79f328a3b8b6ffff04347e; jvm 21.0.3+7-LTS-152
2024-06-01 09:13:29,916 INFO o.e.j.s.h.C.application [main] Initializing Spring embedded WebApplicationContext
2024-06-01 09:13:29,917 INFO o.s.b.w.s.c.ServletWebServerApplicationContext [main] Root WebApplicationContext: initialization completed in 1504 ms
2024-06-01 09:13:30,186 INFO o.e.j.s.DefaultSessionIdManager [main] Session workerName=node0
2024-06-01 09:13:30,204 INFO o.e.j.s.h.ContextHandler [main] Started osbwej.JettyEmbeddedWebAppContext@7f2b39a{application,/,b=[file:/C:/Users/Leopoldo/AppData/Local/Temp/jetty-docbase.8080.17387073418958260011/, URLResource@8043F6F4(jar:nested:///C:/Program%20Files/Stirling-PDF/Stirling-PDF.exe/!BOOT-INF/lib/swagger-ui-5.2.0.jar!/META-INF/resources/)],a=AVAILABLE,h=oeje10s.SessionHandler@760f1081{STARTED}}
2024-06-01 09:13:30,205 INFO o.e.j.e.s.ServletContextHandler [main] Started osbwej.JettyEmbeddedWebAppContext@7f2b39a{application,/,b=[file:/C:/Users/Leopoldo/AppData/Local/Temp/jetty-docbase.8080.17387073418958260011/, URLResource@8043F6F4(jar:nested:///C:/Program%20Files/Stirling-PDF/Stirling-PDF.exe/!BOOT-INF/lib/swagger-ui-5.2.0.jar!/META-INF/resources/)],a=AVAILABLE,h=oeje10s.SessionHandler@760f1081{STARTED}}
2024-06-01 09:13:30,215 INFO o.e.j.s.Server [main] Started oejs.Server@6a2d867d{STARTING}[12.0.7,sto=0] @2865ms
2024-06-01 09:13:30,266 INFO s.s.S.SPdfApplication [main] Running configs ApplicationProperties [security=Security [enableLogin=false, oauth2=null, initialLogin=null, csrfDisabled=true], system=System [defaultLocale=en-US, googlevisibility=false, enableAlphaFunctionality=false, showUpdate=false, showUpdateOnlyAdmin=false], ui=UserInterface [appName=, homeDescription=, appNameNavbar=], endpoints=Endpoints [toRemove=[], groupsToRemove=[CLI]], metrics=Metrics [enabled=true], automaticallyGenerated=AutomaticallyGenerated [key=MASKED], autoPipeline=null]
2024-06-01 09:13:30,742 INFO s.s.S.c.EndpointConfiguration [main] Disabling file-to-pdf
2024-06-01 09:13:30,743 INFO s.s.S.c.EndpointConfiguration [main] Disabling repair
2024-06-01 09:13:30,743 INFO s.s.S.c.EndpointConfiguration [main] Disabling xlsx-to-pdf
2024-06-01 09:13:30,743 INFO s.s.S.c.EndpointConfiguration [main] Disabling ocr-pdf
2024-06-01 09:13:30,743 INFO s.s.S.c.EndpointConfiguration [main] Disabling pdf-to-pdfa
2024-06-01 09:13:30,744 INFO s.s.S.c.EndpointConfiguration [main] Disabling pdf-to-word
2024-06-01 09:13:30,744 INFO s.s.S.c.EndpointConfiguration [main] Disabling html-to-pdf
2024-06-01 09:13:30,744 INFO s.s.S.c.EndpointConfiguration [main] Disabling book-to-pdf
2024-06-01 09:13:30,744 INFO s.s.S.c.EndpointConfiguration [main] Disabling pdf-to-xml
2024-06-01 09:13:30,744 INFO s.s.S.c.EndpointConfiguration [main] Disabling compress-pdf
2024-06-01 09:13:30,745 INFO s.s.S.c.EndpointConfiguration [main] Disabling pdf-to-presentation
2024-06-01 09:13:30,745 INFO s.s.S.c.EndpointConfiguration [main] Disabling url-to-pdf
2024-06-01 09:13:30,745 INFO s.s.S.c.EndpointConfiguration [main] Disabling pdf-to-book
2024-06-01 09:13:30,745 INFO s.s.S.c.EndpointConfiguration [main] Disabling pdf-to-rtf
2024-06-01 09:13:30,746 INFO s.s.S.c.EndpointConfiguration [main] Disabling extract-image-scans
2024-06-01 09:13:30,746 INFO s.s.S.c.EndpointConfiguration [main] Disabling pdf-to-html
2024-06-01 09:13:30,976 INFO o.s.b.a.w.s.WelcomePageHandlerMapping [main] Adding welcome page: ServletContext resource [/index.html]
2024-06-01 09:13:31,741 INFO o.s.b.a.e.w.EndpointLinksResolver [main] Exposing 1 endpoint(s) beneath base path '/actuator'
2024-06-01 09:13:31,827 INFO o.e.j.s.h.C.application [main] Initializing Spring DispatcherServlet 'dispatcherServlet'
2024-06-01 09:13:31,827 INFO o.s.w.s.DispatcherServlet [main] Initializing Servlet 'dispatcherServlet'
2024-06-01 09:13:31,828 INFO o.s.w.s.DispatcherServlet [main] Completed initialization in 1 ms
2024-06-01 09:13:31,866 INFO o.e.j.s.AbstractConnector [main] Started ServerConnector@f0c1ae1{HTTP/1.1, (http/1.1)}{0.0.0.0:8080}
2024-06-01 09:13:31,867 INFO o.s.b.w.e.j.JettyWebServer [main] Jetty started on port 8080 (http/1.1) with context path '/'
2024-06-01 09:13:31,881 INFO s.s.S.SPdfApplication [main] Started SPdfApplication in 3.913 seconds (process running for 4.531)
2024-06-01 09:13:32,899 INFO s.s.S.SPdfApplication [main] Stirling-PDF Started.
2024-06-01 09:13:32,900 INFO s.s.S.SPdfApplication [main] Navigate to http://localhost:8080
```
I checked for the presence of `settings.yml` file and - as can be seen from the log - it was created with its defaults.

At this point I closed both the terminal window and the Firefox browser; I waited for a little bit and I tried to restart Stirling-PDF. What happened in this case was:
- It opened the first terminal
- it asked for the permission to run as administrator
- after my OK, the first terminal window closed and started the second terminal window which closed abruptly

In this case the info.log was the following:
```
2024-06-01 09:15:00,960 ERROR o.s.b.SpringApplication [main] Application run failed
java.lang.RuntimeException: Failed to initialize application configuration
	at stirling.software.SPDF.config.ConfigInitializer.initialize(ConfigInitializer.java:24)
	at org.springframework.boot.SpringApplication.applyInitializers(SpringApplication.java:627)
	at org.springframework.boot.SpringApplication.prepareContext(SpringApplication.java:400)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:333)
	at stirling.software.SPDF.SPdfApplication.main(SPdfApplication.java:100)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:91)
	at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:53)
	at org.springframework.boot.loader.launch.JarLauncher.main(JarLauncher.java:58)
Caused by: java.nio.file.FileSystemNotFoundException: null
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.getFileSystem(ZipFileSystemProvider.java:156)
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.getPath(ZipFileSystemProvider.java:142)
	at java.base/java.nio.file.Path.of(Path.java:209)
	at java.base/java.nio.file.Paths.get(Paths.java:98)
	at stirling.software.SPDF.config.ConfigInitializer.ensureConfigExists(ConfigInitializer.java:49)
	at stirling.software.SPDF.config.ConfigInitializer.initialize(ConfigInitializer.java:22)
	... 9 common frames omitted
```

So it seems not to work  :-(

Regards,
Topoldo/Leopoldo

--- a/src/main/java/stirling/software/SPDF/config/ConfigInitializer.java
+++ b/src/main/java/stirling/software/SPDF/config/ConfigInitializer.java
@@ -1,8 +1,10 @@
 package stirling.software.SPDF.config;
 
+import java.io.BufferedReader;
 import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.InputStreamReader;
 import java.net.URISyntaxException;
 import java.nio.file.Files;
 import java.nio.file.Path;
@@ -45,15 +47,27 @@ public class ConfigInitializer
                 }
             }
         } else {
-            Path templatePath =
-                    Paths.get(
-                            getClass()
-                                    .getClassLoader()
-                                    .getResource("settings.yml.template")
-                                    .toURI());
-            Path userPath = Paths.get("configs", "settings.yml");
+            // Load the template content from classpath
+            List<String> templateLines;
+            try (InputStream in =
+                    getClass().getClassLoader().getResourceAsStream("settings.yml.template")) {
+                if (in == null) {
+                    throw new FileNotFoundException(
+                            "Resource file not found: settings.yml.template");
+                }
+                templateLines = new ArrayList<>();
+                try (var reader = new InputStreamReader(in)) {
+                    try (var bufferedReader = new BufferedReader(reader)) {
+                        String line;
+                        while ((line = bufferedReader.readLine()) != null) {
+                            templateLines.add(line);
+                        }
+                    }
+                }
+            }
 
-            List<String> templateLines = Files.readAllLines(templatePath);
+            // Read the user settings file if it exists
+            Path userPath = Paths.get("configs", "settings.yml");
             List<String> userLines =
                     Files.exists(userPath) ? Files.readAllLines(userPath) : new ArrayList<>();
 
@@ -87,6 +101,7 @@ public class ConfigInitializer
             Files.write(userPath, resultLines);
         }
 
+        // Ensure the custom settings file exists
         Path customSettingsPath = Paths.get("configs", "custom_settings.yml");
         if (!Files.exists(customSettingsPath)) {
             Files.createFile(customSettingsPath);

